### DISCUSSION TITLE
Existing VPCs

### DISCUSSION THREAD

**Discussion by afgallo** [aws/copilot-cli#3419] [Q&A] [answered]
Hi there,

I've got a few VPCs that have already been set up following AWS best practices. It has igs, nacls, nats, etc. I am aware that I can import vpcs and subnets using the `env init` command: https://aws.github.io/copilot-cli/docs/commands/env-init/

In that case, will Copilot create any VPC related resources? I'm not sure if it would create another ig or a different CIDR, etc?

Cheers!

**Comment by efekarakus** [comment, marked as answer]
Hi @afgallo !

> In that case, will Copilot create any VPC related resources?

Nope! In that scenario Copilot won't create any of the following resources:
* Skip VPC
* Skip Route Tables
* Skip internet gateway
* Skip subnets
* Skip NAT gateways

We will just assume that you have configured your VPC correctly and just use your VPC ID and the subnets you provide to place tasks.

However, we will still create the following resources:
- An ALB for "Load Balanced Web Services"
- An environment security group to shared between your services

Hope this helps!

**Reply by afgallo** [reply]
That's excellent, thank you @efekarakus
